src/resources/pulse_counter_esp32.cc

```diff
@@ -75,7 +77,7 @@ class PcntUnitResource : public Resource {
     // https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/pcnt.html#install-pcnt-channel
     // This static assert might hit, even though the code is still OK. Check the documentation if
     // the code from 'master' (as of 2022-07-01) has already made it into the release you are using.
-    static_assert(ESP_IDF_VERSION_MAJOR == 4 && ESP_IDF_VERSION_MINOR == 4,
+    static_assert(ESP_IDF_VERSION_MAJOR == 5 && ESP_IDF_VERSION_MINOR == 0,
```

Comment:
It seems like the new API is available.
I will try to get rid of this static_assert once we have switched.